util/signing/signer/src/lib.rs

```diff
@@ -89,36 +66,60 @@ pub trait VerifierOperations<C: cryptography::Curve> {
 		&self,
 		message: Bytes,
 		signature: Signature,
-		public_key: PublicKey,
-	) -> Result<bool, VerifierError>;
+		public_key: PublicKey<C>,
+	) -> Result<bool, SignerError>;
+}
+
+pub trait VerifierOperationsSync<C: cryptography::Curve> {
+	/// Verifies a signature.
+	fn verify(
+		&self,
+		message: Bytes,
+		signature: Signature,
+		public_key: PublicKey<C>,
+	) -> Result<bool, SignerError>;
 }
 
-pub struct Verifier<O, C>
-where
-	O: VerifierOperations<C>,
-	C: cryptography::Curve,
-{
-	operations: O,
-	_curve_marker: std::marker::PhantomData<C>,
+pub struct SignerConfig;
+
+pub struct KeyManager {
+	pub tobedefined: String,
 }
 
-/// Verifier wraps an implementation of [VerifierOperations] and provides a simple API for verifying signatures.
-impl<O, C> Verifier<O, C>
-where
-	O: VerifierOperations<C>,
-	C: cryptography::Curve,
-{
-	pub fn new(operations: O) -> Self {
-		Self { operations, _curve_marker: std::marker::PhantomData }
+impl KeyManager {
```

Comment:
`KeyManager` with a single `KeyId` type will likely not be particularly useful as each cloud already kind of has their own format.

Reply:
The idea was to hide the cloud provider/real key and define more an application key name. For example, the relayer key id is "relayer" and in the key manager config, the relayer key is associated to the AWS cloud and a AWS key id. 
One of the requirement was that the application shouldn't know which provider manages the key so that the effective key can be changed without rebuilding the application.
So the role of the Key manager is to define this mapping between application key_id and the real key.
Perhaps I should rename to keyname.